shared/actions/login/index.js

```diff
@@ -16,11 +16,57 @@ import {defaultModeForDeviceRoles, qrGenerate} from './provision-helpers'
 import {getCurrentStatus} from '../config'
 import type {Dispatch, GetState, AsyncAction, TypedAction} from '../../constants/types/flux'
 import type {incomingCallMapType, login_recoverAccountFromEmailAddress_rpc,
-  login_login_rpc, login_logout_rpc, device_deviceAdd_rpc} from '../../constants/types/flow-types'
+  login_login_rpc, login_logout_rpc, device_deviceAdd_rpc, login_getConfiguredAccounts_rpc} from '../../constants/types/flow-types'
 import {mobileAppsExist} from '../../util/feature-flags'
-
+import {overrideLoggedInTab} from '../../local-debug'
 let currentLoginSessionID = null
 
+export function navBasedOnLoginState () :AsyncAction {
+  return (dispatch, getState) => {
+    const {config: {status}} = getState()
+
+    // No status?
+    if (!status || !Object.keys(status).length) {
+      dispatch(navigateTo([], loginTab))
+      dispatch(switchTab(loginTab))
+    } else {
+      if (status.loggedIn) { // logged in
+        if (overrideLoggedInTab) {
+          console.log('Loading overridden logged in tab')
+          dispatch(switchTab(overrideLoggedInTab))
+        } else {
+          dispatch(switchTab(devicesTab))
+        }
+      } else if (status.registered) { // relogging in
+        dispatch(getAccounts())
+        dispatch(navigateTo(['login'], loginTab))
```

Comment:
I think the component can be in a loading state. i need to work on it more after this moves forward